File: server/src/Application/TeamBarometer/Models/QuestionModel.cs
﻿using Domain.TeamBarometer.Entities;
using System;

namespace Application.TeamBarometer.Models
{
    public class QuestionModel
    {
        public QuestionModel(Question question)
        {
            Id = question.Id;
            Description = question.Description;
            RedAnswer = question.GetAnswerDescription(Answer.Red);
            GreenAnswer = question.GetAnswerDescription(Answer.Green);
            IsTheCurrent = question.IsTheCurrent;
            IsUpForAnswer = question.IsUpForAnswer;
            AmountOfGreenAnswers = question.GetCountOfTheAnswer(Answer.Green);
            AmountOfYellowAnswers = question.GetCountOfTheAnswer(Answer.Yellow);
            AmountOfRedAnswers = question.GetCountOfTheAnswer(Answer.Red);
        }

        public Guid Id { get; set; }
        public string Description { get; private set; }
        public string RedAnswer { get; private set; }
        public string GreenAnswer { get; private set; }
        public bool IsTheCurrent { get; private set; }
        public int AmountOfRedAnswers { get; private set; }
        public int AmountOfYellowAnswers { get; private set; }
        public int AmountOfGreenAnswers { get; private set; }
        public bool IsUpForAnswer { get; set; }

        public override bool Equals(object obj)
        {
            QuestionModel question = obj as QuestionModel;

            return Id == question?.Id;
        }

        public override int GetHashCode()
        {
            return Id.GetHashCode();
        }
    }
}

File: v0/client/react/src/meeting/models/QuestionModel.js
export default class QuestionModel {
    constructor({id, description, redAnswer, greenAnswer, isTheCurrent, isUpForAnswer, amountOfRedAnswers, amountOfYellowAnswers, amountOfGreenAnswers}) {
        this.id = id;
        this.description = description;
        this.redAnswer = redAnswer;
        this.greenAnswer = greenAnswer;
        this.isTheCurrent = isTheCurrent;
        this.amountOfRedAnswers = amountOfRedAnswers;
        this.amountOfYellowAnswers = amountOfYellowAnswers;
        this.amountOfGreenAnswers = amountOfGreenAnswers;
        this.isUpForAnswer = isUpForAnswer;
    }

    isEqualTo(question) {
        return this.description === question.description;
    }
}